libraries/AP_GPS/AP_GPS_NOVA.cpp

```diff
@@ -257,6 +278,42 @@ AP_GPS_NOVA::process_message(void)
         _new_position = true;
     }
 
+    // 971 for unicore 2042 for novatel
+    if(messageid == 971 || messageid == 2042)
+    {
+        const nova_heading &headingu = nova_msg.data.headingu;
+        if (headingu.solstat == 0) {// have a solution
+            switch (headingu.postype)
+            {
+                case 16:
+                case 17: // psrdiff
+                case 18: // waas
+                case 20: // omnistar
+                case 68: // ppp_converg
+                case 69: // ppp
+                case 32: // l1 float
+                case 33: // iono float
+                case 34: // narrow float
+                case 48: // l1 int
+                case 50: // narrow int
+                {
+                    state.have_gps_yaw = true;
+                    state.gps_yaw_configured = true;
+                    state.have_gps_yaw_accuracy = true;
+                    state.gps_yaw = (float)(headingu.heading);
+                    state.gps_yaw_accuracy = (float)(headingu.hdgstddev);
```

Comment:
Why do these need casts to float? The struct member is float as far as I can see.